src/compiler/sys.ts

```diff
@@ -298,6 +298,55 @@ namespace ts {
                 };
             }
 
+            function createWatchedFileSet() {
+                const watchedDirectories = createFileMap<FileWatcher>();
+                const watchedFiles = createFileMap<(fileName: string, removed?: boolean) => void>();
+                const currentDirectory = process.cwd();
+
+                return { addFile, removeFile };
+
+                function addFile(fileName: string, callback: (fileName: string, removed?: boolean) => void): WatchedFile {
+                    const path = toPath(fileName, currentDirectory, getCanonicalPath);
+                    const parentDirPath = toPath(ts.getDirectoryPath(fileName), currentDirectory, getCanonicalPath);

```

Comment:
`toPath` should take care of it, however the valid point here is that separators in `fileName` might still vary and `getDirectoryPath` expects them to be only forward slashes. I agree with @billti that it will be simplier to just call `const parentDirPath = getDirectoryPath(path)`